assets/prototype/domain/eventEditor/data/queries/priceTypes/test/usePriceTypeForPrice.test.ts

```diff
@@ -0,0 +1,92 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import usePriceTypeForPrice from '../usePriceTypeForPrice';
+import { ApolloMockedProvider } from '../../../../context';
+import { nodes, edge } from './data';
+import { nodes as prices } from '../../prices/test/data';
+import useInitPriceTypeTestCache from './useInitPriceTypeTestCache';
+import { isDefaultPriceType } from '../../../../../shared/predicates/priceTypes/selectionPredicates';
+import { useRelations } from '../../../../../../application/services/apollo/relations';
+
+describe('usePriceTypeForPrice()', () => {
+	const wrapper = ApolloMockedProvider();
+	const existingPrice = prices[0];
+	it('returns null as price type when the cache is empty', () => {
+		const { result } = renderHook(() => usePriceTypeForPrice(existingPrice.id), { wrapper });
+
+		expect(result.current).toBeNull();
+	});
+
+	it('returns null as price type when the price does not exist and the cache is empty', () => {
+		const { result } = renderHook(() => usePriceTypeForPrice('fake-id'), { wrapper });
+
+		expect(result.current).toBeNull();
+	});
+
+	it('returns null as price type when the price and default price type do not exist but the cache is NOT empty', () => {
+		const nonDefaultPriceTypeNodes = nodes.filter((priceType) => !isDefaultPriceType(priceType));
+		const { result } = renderHook(
+			() => {
+				useInitPriceTypeTestCache({ ...edge, nodes: nonDefaultPriceTypeNodes });
+				return usePriceTypeForPrice('fake-id');
+			},
+			{ wrapper }
+		);
+
+		expect(result.current).toBeNull();
+	});
+
+	it('returns the default price type when the price does not exist and the cache is NOT empty', () => {
+		const defaultPriceType = nodes.filter(isDefaultPriceType)[0];
+		const { result } = renderHook(
+			() => {
+				useInitPriceTypeTestCache();
+				return usePriceTypeForPrice('fake-id');
+			},
+			{ wrapper }
+		);
+
+		const { current: cachedDefaultPriceType } = result;
+
+		expect(cachedDefaultPriceType).toEqual(defaultPriceType);
+
+		expect(cachedDefaultPriceType.id).toEqual(defaultPriceType.id);
+
+		expect(cachedDefaultPriceType.name).toEqual(defaultPriceType.name);
+	});
```

Comment:
this is sooooo easy to read, great job 🏅

Reply:
agree, also I like the logic flow of the tests :medal_military: